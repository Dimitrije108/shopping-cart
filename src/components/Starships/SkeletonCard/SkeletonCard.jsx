import styles from "./SkeletonCard.module.css"
// Create a ship card component
export default function SkeletonCard() {
	return (
		<div className={styles.skelCard} data-testid="skelCard">
			<div className={styles.skelImg}></div>
			<div className={styles.infoCont}>
				<div className={styles.skelName}>
					<div className={styles.name}></div>
					<div className={styles.imperial}></div>
					<div className={styles.naboo}></div>
				</div>
				<div className={styles.buyCont}>
					<div className={styles.price}></div>
					<div className={styles.credits}></div>
					<div className={styles.addBtn}></div>
				</div>
			</div>
		</div>
	)
};