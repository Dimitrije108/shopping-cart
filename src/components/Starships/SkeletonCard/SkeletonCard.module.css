.skelCard {
	outline: 1px solid rgb(104, 104, 104);
	
  transition: all 0.3s ease-in-out;
  cursor: pointer;
}

.skelImg {
  width: 600px;
  height: 300px;
}

.infoCont {
  height: 95px;
  margin: 8px;
  outline: 1px solid rgb(104, 104, 104);

  display: flex;
  justify-content: space-between;
}

.skelName {
	margin: 0;
	padding: 0.5rem 0 1rem 0;

	display: flex;
  flex-direction: column;
}

.name {
  width: 240px;
  height: 28px;
  margin: 4px 8px 0 8px;

  outline: 1px solid rgb(104, 104, 104);
}

.imperial {
  width: 150px;
  height: 13px;
  margin: 6px 8px 0 8px;
  outline: 1px solid rgb(104, 104, 104);
}

.naboo {
  width: 125px;
  height: 13px;
  margin: 6px 8px 4px 8px;
  outline: 1px solid rgb(104, 104, 104);
}

.buyCont {
  margin: 0;
	padding: 4px 0;

  display: flex;
  flex-direction: column;
}

.price,
.addBtn {
  width: 120px;
  height: 37px;
  margin: 6px 8px 4px 8px;
  outline: 1px solid rgb(104, 104, 104);
}

.credits {
  width: 50px;
  height: 13px;
  margin: 2px 8px 1px 8px;
  outline: 1px solid rgb(104, 104, 104);

  align-self: flex-end;
}

.skelImg,
.name,
.imperial,
.naboo,
.price,
.credits,
.addBtn {
  background: linear-gradient(90deg, #141414 25%, #1d1d1d 50%, #131313 75%);
  background-size: 200% 100%;
  animation: shimmer 5s infinite;
}



@keyframes shimmer {
  from {
    background-position: 200% 0;
  }

  to {
    background-position: -200% 0;
  }
}
