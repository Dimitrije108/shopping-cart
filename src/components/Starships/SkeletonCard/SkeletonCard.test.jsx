import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import SkeletonCard from "./SkeletonCard"

describe("Skeleton Card component", () => {
	it("render the skeleton card component", () => {
		const skeletonCard = render(
			<MemoryRouter>
				<SkeletonCard />
			</MemoryRouter>
		)

		expect(skeletonCard).toMatchSnapshot();
	});
});