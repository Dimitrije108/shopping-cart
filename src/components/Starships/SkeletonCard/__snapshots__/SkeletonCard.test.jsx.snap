// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Skeleton Card component > skeleton card component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_skelCard_5bb178"
        data-testid="skelCard"
      >
        <div
          class="_skelImg_5bb178"
        />
        <div
          class="_infoCont_5bb178"
        >
          <div
            class="_infoWrapper_5bb178"
          >
            <div
              class="_skelName_5bb178"
            >
              <div
                class="_name_5bb178"
              />
              <div
                class="_imperial_5bb178"
              />
              <div
                class="_naboo_5bb178"
              />
            </div>
            <div
              class="_buyCont_5bb178"
            >
              <div
                class="_price_5bb178"
              />
              <div
                class="_credits_5bb178"
              />
            </div>
          </div>
          <div
            class="_addToCartCont_5bb178"
          >
            <div
              class="_addBtn_5bb178"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_skelCard_5bb178"
      data-testid="skelCard"
    >
      <div
        class="_skelImg_5bb178"
      />
      <div
        class="_infoCont_5bb178"
      >
        <div
          class="_infoWrapper_5bb178"
        >
          <div
            class="_skelName_5bb178"
          >
            <div
              class="_name_5bb178"
            />
            <div
              class="_imperial_5bb178"
            />
            <div
              class="_naboo_5bb178"
            />
          </div>
          <div
            class="_buyCont_5bb178"
          >
            <div
              class="_price_5bb178"
            />
            <div
              class="_credits_5bb178"
            />
          </div>
        </div>
        <div
          class="_addToCartCont_5bb178"
        >
          <div
            class="_addBtn_5bb178"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
